src/error.rs

```diff
@@ -139,3 +139,67 @@ where
 {
     DbErr::Json(s.to_string())
 }
+
+/// An error from unsuccessful SQL query
+#[derive(Error, Debug, PartialEq, Eq)]
+#[non_exhaustive]
+pub enum SqlErr {
+    /// error for inserting a record with a key that already exists in the table
+    #[error("Cannot have record with same key")]
+    UniqueConstraintViolation(),
+    /// error for Foreign key is not primary key
+    #[error("Cannot add non-primary key from other table")]
+    ForeignKeyConstraintViolation(),
```

Comment:
Do we have extra data for each error type? If no, then we can simply remove the `()`.

```suggestion
    /// error for inserting a record with a key that already exists in the table
    #[error("Cannot have record with same key")]
    UniqueConstraintViolation,
    /// error for Foreign key is not primary key
    #[error("Cannot add non-primary key from other table")]
    ForeignKeyConstraintViolation,
```

Reply:
Thanks for all the comments! Integrated the later two into my code.